projects/igniteui-angular/src/lib/tabs/tabs.component.ts

```diff
@@ -35,33 +35,108 @@ export enum TabsType {
 })
 
 export class IgxTabsComponent implements AfterViewInit, OnDestroy {
-    @ViewChildren(forwardRef(() => IgxTabItemComponent)) public tabs: QueryList<IgxTabItemComponent>;
-    @ContentChildren(forwardRef(() => IgxTabsGroupComponent)) public groups: QueryList<IgxTabsGroupComponent>;
 
+    /**
+     * Provides an observable collection of all `IgxTabItemComponent`s.
+     * ```typescript
+     * const tabItems = this.myTabComponent.tabs;
+     * ```
+     */
+    @ViewChildren(forwardRef(() => IgxTabItemComponent))
+    public tabs: QueryList<IgxTabItemComponent>;
+
+    /**
+     * Provides an observable collection of all `IgxTabsGroupComponent`s.
+     * ```typescript
+     * const groupItems = this.myTabComponent.tabs;
+     * ```
+     */
+    @ContentChildren(forwardRef(() => IgxTabsGroupComponent))
+    public groups: QueryList<IgxTabsGroupComponent>;
+
+    /**
+     * Defines the tab header sizing mode. You can choose between `contentfit` or `fixed`.
+     * By default the header sizing mode is `contentfit`.
+     * ```html
+     * <igx-tabs tabsType="fixed">
+     *     <igx-tabs-group label="HOME">Home</igx-tabs-group>
+     * </igx-tabs>
+     * ```
+     */
     @Input('tabsType')
     public tabsType: string | TabsType = 'contentfit';
 
+    /**
+     * Emitted when a tab item is selected.
+     * ```html
+     * <igx-tabs (onTabItemSelected)="itemSelected($event)">
+     *      <igx-tabs-group label="Tab 1">This is Tab 1 content.</igx-tabs-group>
+     *      <igx-tabs-group label="Tab 2">This is Tab 2 content.</igx-tabs-group>
+     * </igx-tabs>
+     * ```
+     * ```typescript
+     * itemSelected(e){
+     *      alert("Tab item was selected!");
```

Comment:
Event args return selected IgxTabItemComponent and IgxTabsGroupComponent so you can use them to show some additional info to show.